### PR TITLE
Alerting: Add filters for AlertList panel

### LINKED ISSUE #35720: Alerting: implement missing filters for alert list panel
**What would you like to be added**:
The new alert list panel is missing some filters:

- by dashboard title
- by labels (old version had tags)
- by datasource 
- by lotex group or namespace (not sure if needed)

**Why is this needed**:

### PR DESCRIPTION
**Pull request by nathanrodman**
**What this PR does / why we need it**:

- Adds datasource filtering
- Converges two different alert state filters for Grafana Managed and Prometheus alert states. There was a recent change introduced that wouldn't show prometheus alerts when expanding a rule in the panel and this should fix that and also make the alert state simpler.
- Refactors filtering by label.

**Which issue(s) this PR fixes**:

Fixes #35720 

**Special notes for your reviewer**:

### CODE REVIEW

**Comment by gillesdemey** [comment]
I just tested it locally and it seems like `dataSourceName` is always defined as `grafana` for grafana managed alerts.

Can you double check and confirm if it does correct filtering for `grafana managed` datasources?

**Review by gillesdemey — CHANGES_REQUESTED** [review]

File: public/app/plugins/panel/alertlist/AlertInstances.tsx [hunk 1]
```diff
@@ -45,22 +44,21 @@ export const AlertInstances = ({ ruleWithLocation, options }: Props) => {
...
-        (options.alertInstanceStateFilter.Error && alert.state === GrafanaAlertState.Error)
+        (options.stateFilter.firing &&
+          (alert.state === GrafanaAlertState.Alerting || alert.state === PromAlertingRuleState.Firing)) ||
+        (options.stateFilter.pending &&
+          (alert.state === GrafanaAlertState.Pending || alert.state === PromAlertingRuleState.Pending)) ||
+        (options.stateFilter.noData && alert.state === GrafanaAlertState.NoData) ||
+        (options.stateFilter.normal && alert.state === GrafanaAlertState.Normal) ||
+        (options.stateFilter.error && alert.state === GrafanaAlertState.Error) ||
+        (options.stateFilter.inactive && alert.state === PromAlertingRuleState.Inactive) ||
+        (options.stateFilter.pending && alert.state === PromAlertingRuleState.Pending)
```
[the remark this answers is not in the document]
Comment (nathanrodman) [reply public/app/plugins/panel/alertlist/AlertInstances.tsx:58]: Yes that appears to be a holdover that can be removed

**Comment by nathanrodman** [comment]
> I just tested it locally and it seems like `dataSourceName` is always defined as `grafana` for grafana managed alerts.
> 
> Can you double check and confirm if it does correct filtering for `grafana managed` datasources?

@gillesdemey You are correct. I'm wondering given that whether it should just be limited to Prometheus/Loki datasources for the datasource filter? The `<DatasourcePicker>` has a `type` prop where you can pass in a list of datasource types. So instead of limiting it to the datasources that support alerting, it could just support ds with type prometheus, loki, or grafana.

**Comment by gillesdemey** [comment]
>  I'm wondering given that whether it should just be limited to Prometheus/Loki datasources for the datasource filter? 

I think that would be the best way to go forward here and get unstuck 👍

**Review by gillesdemey — APPROVED** [review]

**Comment by gillesdemey** [comment]
Awesome, Nathan! 🚀

**Pull request merged**